metadata/make_merged_CDE_table.py

```diff
@@ -0,0 +1,50 @@
+#!/bin/env python
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, Gregory Poore.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# ----------------------------------------------------------------------------
+
+import numpy as np
+import pandas as pd
+import subprocess
+import glob
+
+## Call Bash script to transpose data
+print "Transposing CDE file..."
+subprocess.call("./text_transpose.sh", shell=True)
+
+## Find intersection of headers
+
+lines_list = open('column_names_CDEs.txt').read().splitlines()
+split_lines_list = map(str.split, lines_list)
+intersection_headers_set = set.intersection(*map(set,split_lines_list))
+
+## Convert set to string
+
+intersection_headers_string = list(intersection_headers_set)
+print "The intersection of all CDE columns yields: \n"
+print '\n'.join(intersection_headers_string)
+
+## Extract these columns from CDE files into one dataFrame object
+
+print "\n""Concatenating transposed CDE files..."
+allTransposedFiles = glob.glob("./Transposed*.txt")
+dfIntersect = pd.concat(
+	pd.read_csv(eachFile, 
+		sep='\t', 
+		usecols = intersection_headers_string) 
+	for eachFile in allTransposedFiles)
+
+allTransposedFiles = glob.glob("./Transposed*.txt")
+dfFullJoin = pd.concat(
+	pd.read_csv(eachFile, 
+		sep='\t') 
+	for eachFile in allTransposedFiles)
+
+## Write to csv file
+dfIntersect.to_csv('Intersected_CDEs.csv')
+dfFullJoin.to_csv('Full_Joined_CDEs.csv')
+
+print "Complete! Saved in Intersected_CDEs.csv"
```

Comment:
In this specific line, the py3 change is that print is a function. In
general, the differences are minor with the exception of string handling.
There should be easy to find guides of what the differences are though
online. First good place to start is just to run the script with py3 :)

On Aug 11, 2016 01:19, "Greg Poore" <notifications@github.com> wrote:

> In metadata/make_merged_CDE_table.py
> <https://github.com/biocore/tcga/pull/1#discussion_r74341947>:
>
> > +	pd.read_csv(eachFile,
> > +		sep='\t',
> > +		usecols = intersection_headers_string)
> > +	for eachFile in allTransposedFiles)
> > +
> > +allTransposedFiles = glob.glob("./Transposed*.txt")
> > +dfFullJoin = pd.concat(
> > +	pd.read_csv(eachFile,
> > +		sep='\t')
> > +	for eachFile in allTransposedFiles)
> > +
> > +## Write to csv file
> > +dfIntersect.to_csv('Intersected_CDEs.csv')
> > +dfFullJoin.to_csv('Full_Joined_CDEs.csv')
> > +
> > +print "Complete! Saved in Intersected_CDEs.csv"
>
> @wasade <https://github.com/wasade> @ekopylova
> <https://github.com/ekopylova> This was written in Python 2.7. I'm
> relatively new to Python (previously R/Matlab/Perl/SQL), so I am not
> familiar with the specific syntax differences in porting one to the other.
> Does it have to be ported or can it be left as-is?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/tcga/pull/1/files/0de2130298adbac726c4e3f9459c61fb12254054#r74341947>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAc8st7eLrU5R6U0nK7b7Pjyp2fXqLYIks5qek6BgaJpZM4JgsdH>
> .
>